packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts

```diff
@@ -71,20 +80,49 @@ export class QueryDataViewService {
       this.subscriptions
     );
 
-    const webview = this.currentPanel.webview;
-    webview.html = this.getWebViewContent(webview);
+    this.currentPanel.webview.html = this.getWebViewContent(
+      this.currentPanel.webview
+    );
+
+    this.currentPanel.webview.onDidReceiveMessage(
+      this.onDidRecieveMessageHandler,
+      this,
+      this.subscriptions
+    );
 
-    this.updateWebviewWith(webview, this.queryData);
-    return webview;
+    return this.currentPanel.webview;
   }
 
-  private getWebViewContent(webview: vscode.Webview): string {
-    let _html: string;
+  protected onDidRecieveMessageHandler(message: DataViewEvent) {
+    const { type, format } = message;
+    switch (type) {
+      case 'activate':
```

Comment:
This will address a race condition when loading the webview and the table ui, similar to how we do it with the soql builder UI.